### DISCUSSION TITLE
NATS as messeging broker is not yet fully supported

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#1584] [Dev] [answered]
By accident, I realized that NATS as a messaging broker is not yet fully supported.
For example, it is not added as messaging format (`registerMessagingFormats(...)`)  in the archetypes and several extension modules.
Is this just a left-over or is there a specifiy reason for that?
What needs to be done here? simply add NATS or is there some more implementation effort to be spent?

cc @RobertIndie (this might also be interesting for the pulsar integration)

**Comment by RobertIndie** [comment]
While implementing the Pulsar message protocol. I also find that the NATS is not working. There are some steps that are not completed. I'm afraid the NATS deployment is not working at all.

**Reply by dominikriemer** [reply]
Nats is registered in the `AllExtensionsInit` class of extensions-all-jvm and should work for the pipeline elements that are provided from this module (we have a deployment running on Nats).
What's the error you get when using Nats?

**Comment by bossenti** [comment, marked as answer]
I merged in https://github.com/apache/streampipes/pull/1628 which should resolve the initial discussion post
